src/dialogs/more-info/controls/more-info-weather.ts

```diff
@@ -13,26 +13,108 @@ import {
   PropertyValues,
   nothing,
 } from "lit";
-import { customElement, property } from "lit/decorators";
+import { customElement, property, state } from "lit/decorators";
+import { HassEntityBase } from "home-assistant-js-websocket";
 import { formatDateWeekdayDay } from "../../../common/datetime/format_date";
 import { formatTimeWeekday } from "../../../common/datetime/format_time";
 import { formatNumber } from "../../../common/number/format_number";
 import "../../../components/ha-svg-icon";
 import {
+  getForecast,
   getWeatherUnit,
   getWind,
-  isForecastHourly,
+  subscribeForecast,
+  ForecastEvent,
   WeatherEntity,
+  WeatherEntityFeature,
   weatherIcons,
 } from "../../../data/weather";
 import { HomeAssistant } from "../../../types";
+import { supportsFeature } from "../../../common/entity/supports-feature";
 
 @customElement("more-info-weather")
 class MoreInfoWeather extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
   @property() public stateObj?: WeatherEntity;
 
+  @state() private _forecastEvent?: ForecastEvent;
+
+  @state() private _subscribed?: Promise<() => void>;
+
+  private _needForecastSubscription() {
+    return (
+      supportsFeature(
+        this.stateObj as HassEntityBase,
+        WeatherEntityFeature.FORECAST_DAILY
+      ) ||
+      supportsFeature(
+        this.stateObj as HassEntityBase,
+        WeatherEntityFeature.FORECAST_HOURLY
+      ) ||
+      supportsFeature(
+        this.stateObj as HassEntityBase,
+        WeatherEntityFeature.FORECAST_TWICE_DAILY
+      )
+    );
+  }
+
+  private _forecastType() {
+    if (
+      supportsFeature(
+        this.stateObj as HassEntityBase,
+        WeatherEntityFeature.FORECAST_DAILY
+      )
+    ) {
+      return "daily";
+    }
+    if (
+      supportsFeature(
+        this.stateObj as HassEntityBase,
+        WeatherEntityFeature.FORECAST_HOURLY
+      )
+    ) {
+      return "hourly";
+    }
+    if (
+      supportsFeature(
+        this.stateObj as HassEntityBase,
+        WeatherEntityFeature.FORECAST_TWICE_DAILY
+      )
+    ) {
+      return "twice_daily";
+    }
+    return undefined;
+  }
+
+  private _handleForecastEvent(forecastEvent: ForecastEvent) {
+    this._forecastEvent = forecastEvent;
+  }
+
+  private async _subscribeForecastEvents() {
```

Comment:
We should store the entity ID and forecast type of the current subscription. That way we don't have to re-subscribe if just the temperature changes in the state.

Reply:
We only subscribe if we need to subscribe and have not yet done so, there should not be resubscribe on state changes: https://github.com/home-assistant/frontend/blob/5dd7525da1fc540e2ebbe67710cbe910f23ef35b/src/dialogs/more-info/controls/more-info-weather.ts#L141